Add hidesBottomBarWhenPushed in ViewControllerProtocol

ykkd opened the pull request:
- Adding `hidesBottomBarWhenPushed` in ViewControllerProtocol
  - The aim is to add a new rule to IBLinter

kateinoigakukun reviewed (commented):
Thank you for your PR :)
Could you add a test case to ensure that the added field is parsed correctly from an IB file. (You can follow a way that other fields do.)

ykkd commented:
@kateinoigakukun 
Thank you for your review!

I added a storyboard file with 2 UIViewControllers in it.
One with `HidesBottomBarWhenPushed = true` and the other with `false`
and validated the values in the test case I just added.

kateinoigakukun reviewed (approved):
Seems legit 👍
